### ISSUE TITLE
okio.Timeout.throwIfReached(Timeout.java:146) exception when same network call is executed multiple times.

### ISSUE THREAD

**Issue by shabarishbk** [square/okhttp#3382]
Getting thew below exception 

W/System.err:     at okio.Timeout.throwIfReached(Timeout.java:146)
W/System.err:     at okio.Okio$1.write(Okio.java:75)
W/System.err:     at okio.AsyncTimeout$1.write(AsyncTimeout.java:179)
W/System.err:     at okio.RealBufferedSink.flush(RealBufferedSink.java:216)
W/System.err:     at okhttp3.internal.http1.Http1Codec.finishRequest(Http1Codec.java:166)
W/System.err:     at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:64)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
W/System.err:     at com.facebook.stetho.okhttp3.StethoInterceptor.intercept(StethoInterceptor.java:59)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
W/System.err:     at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
W/System.err:     at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
W/System.err:     at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
W/System.err:     at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
W/System.err:     at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:212)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
W/System.err:     at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:179)
W/System.err:     at okhttp3.RealCall.execute(RealCall.java:63)
W/System.err:     at retrofit2.OkHttpCall.execute(OkHttpCall.java:174)
W/System.err:     at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall.execute(ExecutorCallAdapterFactory.java:89)

when my code is executed some 7 or 8 times in succession : 

private List<Repo> getServerRepo() throws Throwable {
        List<Repo> repositories = null;
        retrofit2.Call<Repositories> call = syncApi.listRepositories(getUserUid(), getHeadersForRepository());
        Response<Repositories> repoList;
        try {
            repoList = call.execute();
            repositories = repoList.body().getRepository();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return repositories;
    }

I am using retrofit2.2.0 which uses okhttp 3.8.0 and using simpleXmlConverter for converting XML responses.

Setting up retofit like this : 

new OkHttpClient.Builder().
                addInterceptor(loggingInterceptor)
                //.addInterceptor(new UnzippingInterceptor())
                .addNetworkInterceptor(new StethoInterceptor())
                .connectTimeout(30, TimeUnit.SECONDS)
                .readTimeout(300, TimeUnit.SECONDS)
                .writeTimeout(300,TimeUnit.SECONDS)
                .connectionPool(new ConnectionPool(10, 5, TimeUnit.MINUTES))
                .build();

Retrofit.Builder builder = new Retrofit.Builder();
        builder.baseUrl(baseUrl());
        builder.addConverterFactory(
                new AnnotationTypeConverterFactory(GsonConverterFactory.create(),
                        SimpleXmlConverterFactory.create()));
        return builder.build();

**Comment by JakeWharton** [comment]
Looks like a normal timeout. What makes you think it's a bug?

**Comment by shabarishbk** [comment]
i am getting proper response from server for the previous calls, in my scenario that particular call happens multiple times in succession in a mulitthread environment and when execute happens immediately this exception is thrown.

**Comment by swankjesse** [comment]
Unsure what action to take here. If you could isolate this into a test case, that’ll allow us to do something about it!

**Comment by sedstrom** [comment]
I am seeing this too. I make multiple (successful) calls to the same endpoint and will eventually run into this problem. Could be related to RX. Perhaps threads are being terminated and re-used at the same time.

**Comment by ridcully99** [comment]
I'm seeing this too -- strange thing: when I inspect the requests via StethoInterceptor (i.e. really watch them in Chrome) the timeouts go away and it works fine.

**Comment by lukesterlee** [comment]
I'm seeing this too. started to get crashes after upgrading to rxjava2

**Comment by ridcully99** [comment]
We managed to get rid of the issue. I will describe what we did, maybe it helps someone.

What we had:
Multiple Completables (A,B,C), each doing network requests, linked with andThen() to run one after the other. In one Completable (B) we used our own Scheduler implementation that always provides the same single Thread ("single-thread-B"). The Scheduler was shut down (which deletes the thread) in B.doOnTerminate(). 
The exception happend, because it turned out, that Completable C (although told to use Schedulers.io()) ran on our "single-thread-B" somehow, and when the Scheduler got shut down, it crashed.

What we did:
In our Completable B, we changed shutting down our Scheduler in doOnTerminate() to shutting it down in doFinally(), and this solved the issue.
